### ISSUE TITLE
Support for MySQLdb module

### ISSUE THREAD

**Issue by neocogent** [gordonc/gevent-db#2]
I tried for some time to get this module to work but could not figure out a connection string. 

In the end I made a small change to allow it to accept a tuple/list and pass that down to the module. 

On line 53, I simply added the * to have it break out args.

        self.conn = self.apply(__import__(modulename).connect,*connectionstring)

Probably by adding something like isinstance(connectionstring, list) this could be made more generic. Anyway, the above is working for me now and happily I'm using your little pool code in my gevent.pywsgi based server.

Also, I added a __getitem__ function to the cursor so that it supports iteration. This is something the MySQLdb module provides and I was using it in my code. I'm not sure if other modules usually provide this or if it's an extension of the standard. Here's my very simple code if you want to add that ability (useful).

In DBCursor,

```
def __getitem__(self,*args):
        result = self.fetchone()
        if not result:
            raise IndexError
        return result
```

**Comment by neocogent** [comment]
I found this worked for me:

    *(connectionstring,) if isinstance(connectionstring, basestring) else connectionstring

as parameter for connect. It worked as string for the testunit, but also as list for MySQLdb.

**Comment by gordonc** [comment]
wow, havent looked at this in a while. I agree with your first point on the connectionstring ... looking at PEP 0249, the signature is just connect ( parameters... ). On your second point, I think the most straight forward solution is to just expose the underlying cursor functionality.

would this work for your use case?

    def __iter__(self,*args):
        return self.conn.apply(self.cursor.__iter__,*args)

**Comment by neocogent** [comment]
Thanks for responding.
Yes, that worked fine for me and is a better idea. I've updated my copy.

I also made another couple edits on my copy to support the "with" context manager (as does MySQLdb). I find this useful for doing a query (or few) without tying up a connection for longer. I'm not sure I saw a way to explicitly release a "get", so this helped me. If more lengthy processing is being done in a function it doesn't have to hold that resource until exited. Maybe there's a better way or this can also be exposed as with __iter__.

    def __enter__(self):
        return self.cursor
    def __exit__(self, type, value, traceback):
        pass

Allows me to do,

    with dbpool.get().cursor() as cur:
        cur.execute(...)

And have it release after use, or at least I think it does.
